### PR TITLE
Component version telemetry initializer and associated functional test changes.

### PR DESCRIPTION
**Pull request by kartang**

### CODE REVIEW

File: src/Microsoft.ApplicationInsights.AspNet/TelemetryInitializers/ComponentVersionTelemetryInitializer.cs [hunk 1]
```diff
@@ -1,42 +1,45 @@
...
     using DataContracts;
     using Microsoft.AspNet.Http;
     using Microsoft.Extensions.Configuration;
-    using Microsoft.ApplicationInsights.Extensibility;
 
     /// <summary>
     /// A telemetry initializer that populates telemetry.Context.Component.Version to the value read from project.json
     /// </summary>
-    public class ComponentVersionTelemetryInitializer : ITelemetryInitializer
+    public class ComponentVersionTelemetryInitializer : TelemetryInitializerBase
```
Comment (dnduffy) [inline src/Microsoft.ApplicationInsights.AspNet/TelemetryInitializers/ComponentVersionTelemetryInitializer.cs]: You don't appear to use any extra functionality from TelemetryInitializerBase so you can stick with just the interface ITelemetryInitializer.

**Comment by dnduffy** [comment]
:shipit:

File: src/Microsoft.ApplicationInsights.AspNet/TelemetryInitializers/ComponentVersionTelemetryInitializer.cs [hunk 2]
```diff
@@ -1,42 +1,45 @@
...
+                    var config = new ConfigurationBuilder()
+                        .AddJsonFile("project.json")
+                        .Build();
+                    this.appVersion = config[_versionConfigurationOption];
+                    this.isConfigBuilt = true;
+                }
+            }
+            catch(Exception e)
+            {
+                // Add logging through event source. We may not want to through the exception here?
```
Comment (AlexBulankou) [inline src/Microsoft.ApplicationInsights.AspNet/TelemetryInitializers/ComponentVersionTelemetryInitializer.cs]: So we can add it now?

Comment (kartang) [inline src/Microsoft.ApplicationInsights.AspNet/TelemetryInitializers/ComponentVersionTelemetryInitializer.cs]: Will do that and merge!

Comment (kartang) [inline src/Microsoft.ApplicationInsights.AspNet/TelemetryInitializers/ComponentVersionTelemetryInitializer.cs]: Updated the code! Please let me know if that looks ok now!

**Pull request merged**